src/time_steppers/backwardEuler.cpp

```diff
@@ -165,6 +166,63 @@ void backwardEuler::lineSearch(double dt) {
 }
 
 
+void backwardEuler::lineSearch_wolfe(double dt) {
```

Comment:
Like before, instead of multiple functions, let's have the following structure:

an abstract base class `LineSearch`
And then `WolfeLineSearch` and `GoldSteinLineSearch` has child classes.

Reply:
Do we really need to define a class? I think an input parameter of the line search function might be enough.

Reply:
I think having a separate class for the linesearch is cleaner for the following reasons:
1) It abstracts away the linesearch logic from `backwardEuler`. Ideally, `backwardEuler` should only contain the logic necessary for itself. Even now, we have variables like `x_ls` in `backwardEuler`, which is a bit messy.
2) It allows us to scale to larger number of linesearch methods in the future. Imagine we keep implementing different line searches in the future, in that case we don't want N different implementations in `backwardEuler`.